worker/spacenamer/manifold.go

```diff
@@ -0,0 +1,63 @@
+// Copyright 2019 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package spacenamer
+
+import (
+	"gopkg.in/juju/worker.v1"
+	"gopkg.in/juju/worker.v1/dependency"
+
+	"github.com/juju/juju/agent"
+	"github.com/juju/juju/api/base"
+	"github.com/juju/juju/api/spacenamer"
+)
+
+// Logger represents a loggo logger for the purpose of recording what is going
+// on.
+type Logger interface {
+	Debugf(string, ...interface{})
+	Infof(string, ...interface{})
+	Warningf(string, ...interface{})
+	Errorf(string, ...interface{})
+}
+
+// ManifoldConfig defines the names of the manifolds on which a
+// Manifold will depend.
+type ManifoldConfig struct {
+	AgentName     string
+	APICallerName string
+	Logger        Logger
+	NewWorker     func(WorkerConfig) (worker.Worker, error)
+	NewClient     func(base.APICaller) SpaceNamerAPI
```

Comment:
I think you can remove this, instantiate the client directly in `Manifold` and remove the shim.